src/Forum/ForumServiceProvider.php

```diff
@@ -204,8 +205,11 @@ protected function setDefaultRoute(RouteCollection $routes)
         $factory = $this->container->make(RouteHandlerFactory::class);
         $defaultRoute = $this->container->make('flarum.settings')->get('default_route');
 
-        if (isset($routes->getRoutes()['GET'][$defaultRoute]['handler'])) {
-            $toDefaultController = $routes->getRoutes()['GET'][$defaultRoute]['handler'];
+        $dispatcher = new Dispatcher\GroupCountBased($routes->getRouteData());
+        $routeInfo = $dispatcher->dispatch('GET', $defaultRoute);
+
+        if ($routeInfo[0] === Dispatcher::FOUND) {
+            $toDefaultController = $routeInfo[1]['handler'];
```

Comment:
We could also just switch back to using `getRouteData()[0]` instead of `getRoutes()`, it was initially changed to `getRoutes()` because at first, routes were only applied once, but then we changed the implementation to be able to apply them multiple times, but we didn't revert using `getRoutes()`.

Reply:
Ah thats fair. A benefit of this approach is that it could allow setting a discussion directly as the homepage (and support including it in the payload), but no query params would be passed, so unless that were to be implemented, it would just break.

I'll revert back to `getRouteData()[0]` for now, we have other priorities this cycle.